crates/service-client/src/request_identity/v1.rs

```diff
@@ -58,14 +57,26 @@ impl SigningKey {
     }
 }
 
-pub struct Signer<'key> {
-    claims: Claims,
+#[derive(Debug, thiserror::Error)]
+pub enum SigningPrivateKeyReadError {
+    #[error("Only one private key in PEM format is expected, found {0}")]
+    OneKeyExpected(usize),
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+    #[error(transparent)]
+    Pem(#[from] pem::PemError),
+    #[error("Key was rejected by ring: {0}")]
+    KeyRejected(ring::error::KeyRejected),
+}
+
+pub struct Signer<'key, 'aud> {
+    claims: Claims<'aud>,
     signing_key: &'key SigningKey,
 }
 
 #[derive(Serialize, Deserialize)]
-pub(crate) struct Claims {
-    aud: String,
+pub(crate) struct Claims<'aud> {
+    aud: Cow<'aud, str>,
```

Comment:
Are there any follow-up plans with this `Cow`? At the moment it seems to only hold borrowed data. If this should not change, then one could also `aud: &'aud str` which is a tad bit easier.

Reply:
When unmarshalling eg in tests it owns the data. I could unmarshal into a hashmap if easier though

Reply:
Ah, I see. It's because jsonwebtoken::decode requires a `DeserializeOwned` type, right? Thanks for the clarification. Makes sense now for me now. Maybe add a comment stating this fact.